src/debugging/DockerDebugConfigurationProvider.ts

```diff
@@ -26,7 +26,21 @@ export interface DockerAttachConfiguration extends NetCoreDockerDebugConfigurati
 export class DockerDebugConfigurationProvider implements DebugConfigurationProvider {
     public constructor(
         private readonly helpers: { [key in DockerPlatform]: DebugHelper }
-    ) { }
+    ) {
+        // Listen for debug termination events to shut down debug containers as needed
+        registerEvent('debugTermination', debug.onDidTerminateDebugSession, async (context: IActionContext, session: DebugSession) => {

```

Comment:
Just one question: should we un-register these events when the extension is deactivated?

Reply:
That actually does happen since `registerEvent` pushes it onto `extensionContext.subscriptions`: https://github.com/microsoft/vscode-azuretools/blob/main/ui/src/registerEvent.ts

Reply:
Perfect!